lib/robots/sdr_repo/preservation_ingest/update_catalog.rb

```diff
@@ -33,13 +33,36 @@ def handler
         end
 
         def update_catalog
+          wait_as_needed # give Ceph MDS some breathing room on the pres cat side, see comment on method
           with_retries(max_tries: 3, handler: handler, rescue: Preservation::Client::Error) do
             Preservation::Client.update(druid: druid,
                                         version: moab_object.current_version_id,
                                         size: moab_object.size,
                                         storage_location: moab_object.storage_root)
           end
         end
+
+        # sleep for a configured amount of time, including a random additional delay to help ensure staggering.
+        #
+        # BUT WHY? at present we're having an issue with Ceph backed preservation storage that plays out something like:
+        # * a preservation_robots worker finishes writing a new version of a Moab to its druid tree in the storage trunk, from a bag
+        #   left in the deposit trunk.
+        # * the preservation_robots worker signals preservation_catalog that it has finished updating the Moab (update_catalog, above).
+        # * preservation_catalog, running on a different VM from the preservation_robots worker, attempts to read the Moab contents.
+        #   * unfortunately, the Ceph MDS (Metadata Server) that responds to a preservation_catalog VM sometimes lags in learning that the
+        #     preservation_robots VM that wrote the Moab has given up the original write lock, and then preservation_catalog effectively
+        #     has trouble fully reading the contents of the new Moab version.  In our experience, a locked file will sometimes not become
+        #     available to pres cat until the MDS instance is restarted.  Though sometimes it becomes available on its own tens of seconds later.
+        #   * usually this happens when trying to read and zip the Moab version for cloud replication in a later async pres cat job.  but sometimes
+        #     it happens when the more cursory version check on the pres cat side of the update_catalog API call is executed.
+        #   * so there's an argument for putting the delay on the other side, in the API implementation.  but putting a multi-second delay
+        #     in something that's already an asynchronous job feels less icky than putting it in a REST method implementation, and there's
+        #     inescapable interplay here between pres robots and pres cat anyway.
+        # we hope to figure out a way to further tune our Ceph setup so that this delay is no longer needed.
+        def wait_as_needed
+          additional_delay = rand(0.0..Settings.hacks.update_catalog_sleep.max_stagger)
```

Comment:
`rand` seemed sufficient here, since i just wanted a crude additional assurance of staggering, beyond the natural staggering of jobs naturally finishing at different times.  i wasn't sure it was necessary, but it didn't seem harmful.

`SecureRandom` felt like overkill, but i'm happy to switch to that if there's a reason.  i tested `SecureRandom.random_number` on the console, and it accepted the same args as the `Kernel.rand` call above, producing similarly bounded random floating point results (presumably with genuine randomness instead of the seeded psuedo-randomness of the current approach).

Reply:
I doubt we need the stagger - I think the issue of staggering isn't one we felt was necessary elsewhere, so why do you want to introduce it here?   I agree that it doesn't harm anything, but it makes the solution feel overengineered to me.